SyncTests/RecordTests.swift

```diff
@@ -47,9 +47,14 @@ class RecordTests: XCTestCase {
         let invalidPayload = "{\"id\": \"abcdefghijkl\", \"collection\": \"clients\", \"payload\": \"invalid\"}"
         let emptyPayload = "{\"id\": \"abcdefghijkl\", \"collection\": \"clients\", \"payload\": \"{}\"}"
 
+        let invalidIDPayload: [String: Any] = ["id": 0]
```

Comment:
I renamed this to "badPayloadGUIDPayload", and added a comment.

I also added a sibling check for an altogether missing GUID.